src/user/room.ts

```diff
@@ -428,12 +428,31 @@ export class Room extends Accessory implements Sayable {
     textOrContactOrFileOrUrl : string | Contact | FileBox | UrlLink,
     mention?                 : Contact | Contact[],
   ): Promise<void> {
+
+    const getRoomAlias: (c: Contact) => Promise<string> =
+    async (c: Contact) => {
+      try {
+        const alias: string | null = await c.alias()
+        return alias ? alias : c.name()
+      } catch (e) {
+        log.error('Room', 'say(%s,%s) get menction roomAlias error',
+                                        textOrContactOrFileOrUrl,
+                                        c.name())
+        return c.name()
+      }
+    }
+
+    const mentionAlias: string = Array.isArray(mention)
+                                  ? mention.map(async (c) => {
+                                      const roomAlias = await getRoomAlias(c)
```

Comment:
How about let's save `mention` to a `mentionContactList` like:

```
const mentionContactList = [...mention]
```

Then could we use 

```
const mentionNameList = await Promise.all(
  mentionContactList.map(
    c => this.alias(c) || c.name()
  )
)
``` 

instead of calling `getRoomAlias(c)`?

Reply:
but if we using Promise.all() we might get (string|null) array for mentionNameList right?

Reply:
![image](https://user-images.githubusercontent.com/25014776/46192806-2628ad00-c2b1-11e8-9a31-064fde889a29.png)

Reply:
Yes, so you need another `.filter()` to filter those `null` values out.

Reply:
Yes, I did try that...but ..
![image](https://user-images.githubusercontent.com/25014776/46197847-cbe31880-c2bf-11e8-8123-8f1e556a7833.png)

Reply:
Yes, you can not rely on my pseudo code: you have to write it by yourself, good luck!